tests/tracking/_tracking_service/test_utils.py

```diff
@@ -371,6 +373,22 @@ def test_set_tracking_uri_with_path(tmp_path, monkeypatch, absolute):
         assert get_tracking_uri() == path.absolute().resolve().as_uri()
 
 
+def test_set_tracking_uri_update_trace_provider():
+    default_uri = mlflow.get_tracking_uri()
+
+    mlflow.tracing.enable()
+    assert get_tracer_tracking_uri() != "file:///tmp"
+
+    set_tracking_uri("file:///tmp")
+    assert get_tracer_tracking_uri() == "file:///tmp"
+
+    set_tracking_uri("https://foo")
+    assert get_tracer_tracking_uri() == "https://foo"
+
+    # clean up
+    set_tracking_uri(default_uri)
```

Comment:
Can we use `try-finally` to ensure that `set_tracking_uri(default_uri)` is always executed?